Vizzuality/marxan-cloud
# Limiting uploads file size (frontend) + friendlier file size error me…

This PR limits (updates the limits) of the uploads size. 

### Overview

- An `util` file was added to help with units conversion, currently only `bytesToMegabytes`  
- A `constants` file was added, where all uploader limits are set together, making it easier to change them in the future   
- Frontend error messages relative to the max upload size being exceeded, currently returned by `react-dropzone`, were tweaked.  
    Unfortunately `react-dropzone` returns something like _"File is larger than 1000000 bytes"_, which is not very intuitive both because the user may not be familiar with what a `byte` is, but also because our recommended file size is displayed in `Megabytes`. 
    Now the error message will be _"File is larger than 10 MB"_. 
- Specific error messages returned by the backend will now be displayed in the "toast", instead of just a _"(...) could not be uploaded"_   

Updated limits:  
- Planning units: 1MB  
- Planning areas: 1MB   
- Planning areas with grid: 10MB  
- Protected areas: 10MB  
- Features: 20MB  
- Cost Surface: 20MB  

### Testing instructions

Verify that the respective uploaders display the correct limits, and that when attempting to upload a file which exceeds the limits, an error message is displayed saying _"File is larger than X MB."_  

Increase the size limits in `constants/file-uploader-size-limits.js` to bypass the file size frontend validation and trigger a backend error, and verify that the resulting toast error message displays the backend error message. 

Files with arbitrary sizes to test can be generated with: https://pinetools.com/random-file-generator

### Feature relevant tickets

[MARXAN-1065](https://vizzuality.atlassian.net/browse/MARXAN-1065)
[MARXAN-1170](https://vizzuality.atlassian.net/browse/MARXAN-1170)
